lib/pure/strutils.nim

```diff
@@ -154,6 +154,59 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string): bool =
+  ## Check whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ## Note: The reason why `parseFloat()` is not used to achieve this is its
+  ## poor performance.
```

Comment:
Also, can you back up your claim about poor performance of `parseFloat` with some numbers? Can we improve it?